src/app/core/component/rating/rating.component.ts

```diff
@@ -9,21 +12,99 @@ import { Rating } from '../../model/rating.enum';
 export class RatingComponent implements OnInit {
 
   @Input() row = true;
-  @Input() rating: number;
-  @Input() userRatingPast: number;
-  @Output() userRatingCurrent = new EventEmitter();
+  @Input() disabled = false;
 
-  constructor() { }
+  @Input() upVotes: string[] = [];
+  @Input() downVotes: string[] = [];
+  @Input() ratingEntity: any;
+  @Input() context: number;
+  @Input() commentEntity: PAComment;
+
+  // primary, accent, warn, ''
+  colorUp = 'primary'
+  colorDown = 'primary'
+
+
+  constructor(
+    private ratingService: RatingManagementService,
+    public auth: AuthenticationService
+  ) { }
 
   ngOnInit(): void {
+    this.auth.user.subscribe(_user => {
+      if(_user && this.upVotes && this.downVotes) {
+        if (this.upVotes.includes(_user.id)) this.setButtonColor(1);
+        else if (this.downVotes.includes(_user.id)) this.setButtonColor(-1);
+        else this.setButtonColor(0);
+      }
+    })
   }
 
-  up() {
-    this.userRatingCurrent.emit(Rating.UP);
+  update(ratingModel: RatingModel) {
+    this.setButtonColor(ratingModel.rating);
+    if (ratingModel.rating == 1) {
+      this.upVotes.push(ratingModel.userId);
+      // CASE DOWN -> UP
+      const index = this.downVotes.indexOf(ratingModel.userId);
+      if (index > -1) this.downVotes.splice(index, 1);
+    }
+    if (ratingModel.rating == -1) {
+      this.downVotes.push(ratingModel.userId);
+      // CASE UP -> DOWN
+      const index = this.upVotes.indexOf(ratingModel.userId);
+      if (index > -1) this.upVotes.splice(index, 1);
+    }
+    if (ratingModel.rating == 0) {
+      const indexUp = this.upVotes.indexOf(ratingModel.userId);
+      if (indexUp > -1) this.upVotes.splice(indexUp, 1);
+      const indexDown = this.downVotes.indexOf(ratingModel.userId);
+      if (indexDown > -1) this.downVotes.splice(indexDown, 1);
+    }
   }
 
-  down(){
-    this.userRatingCurrent.emit(Rating.DOWN);
-  } 
+  click(rating: number) {
+    switch (this.context) {
```

Comment:
what do the number of context mean exactly? could you use an enum instead?

Reply:
Added an enum [Context](https://github.com/PatternAtlas/pattern-atlas-ui/blob/4c29ad8d9d9cb333488ba9c066ff8f33717d9893/src/app/core/shared/_enums/context.enum.ts)